runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -383,7 +382,34 @@ private boolean isTrimCoveredByCheckpoint(long trimMark) {
      * @return whether this stream is capable of being checkpointed
      */
     private boolean isCheckpointCapable() {
-        return !getId().equals(ObjectsView.TRANSACTION_STREAM_ID);
+        return !getId().equals(ObjectsView.TRANSACTION_STREAM_ID)
+                && getStreamOptions().isCheckpointCapable();
+    }
+
+    @Override
+    protected long getMaxGlobalFromMaxEntries(int maxEntries) {
+        if (maxEntries == Integer.MAX_VALUE) {
+            return Address.MAX;
+        }
+
+        StreamAddressSpace streamAddressSpace = runtime.getSequencerView()
+                .getStreamAddressSpace(new StreamAddressRange(getId(), Address.MAX, getCurrentGlobalPosition()));
+
+        long size = streamAddressSpace.getAddressMap().getLongCardinality();
+        if (size == 0L) {
+            return Address.NON_ADDRESS;
+        }
+
+        if (size <= maxEntries) {
+            return streamAddressSpace.getHighestAddress();
+        }
+
+        LongIterator it = streamAddressSpace.getAddressMap().getLongIterator();
+        while (--maxEntries > 0) {
+            it.next();
+        }
+
+        return it.next();
```

Comment:
You can simplyf this logic to 

```java
        long lastAddress = 0;
        while (iterator.hasNext() && --maxEntries > 0) {
            lastAddress = iterator.next();
        }

        return lastAddress;
```

Reply:
The first block is an optimization to avoid iteration if maxEntries is greater than the size of the bitmap, so we can directly return the highest address.